pkg/expansion/system.go

```diff
@@ -8,14 +8,24 @@ import (
 	"sync"
 
 	expansionunversioned "github.com/open-policy-agent/gatekeeper/apis/expansion/unversioned"
+	"github.com/open-policy-agent/gatekeeper/pkg/logging"
 	"github.com/open-policy-agent/gatekeeper/pkg/mutation"
 	mutationtypes "github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-var ExpansionEnabled *bool
+var (
+	ExpansionEnabled *bool
+	log              = logf.Log.WithName("expansion").WithValues(logging.Process, "expansion")
+)
+
+// maxRecursionDepth specifies the maximum call depth for recursive expansion.
+// Theoretically, it should be impossible for a cycle to be created but this
+// measure is put in place as a safeguard.
+const maxRecursionDepth = 30
```

Comment:
should we start even smaller? like 5? thinking about how we can prevent users from doing something too destructive at first with this

Reply:
I was thinking of this is more of a safety-net in case there's a bug that allows cycles, as opposed to something to curb bad actors. 

One could always define a very "breadth-y" graph (e.g. one GVK gets expanded into 500 different types) if they purely wished to hog resources.